### ISSUE TITLE
Fingerprint not saving

### ISSUE THREAD

**Issue by jxosborne** [Kunzisoft/KeePassDX#149]
I've gone through the previous bugs related to getting fingerprint to work, I don't think any apply.  When I enter my password, then scan my finger to store the print, the scan is successful(the phone vibrates, and the text "No password stored yet for this database" disappears.  But when I exit the program it never saves my print.  the text returns to the above as if I've never scanned my fingerprint.

I've tried deleting decryption key, wiping app data, nothing seems to work.  Any ideas?

**Comment by J-Jamet** [comment]
In your case, it must surely come from the keystore that empties automatically, it may be a system incompatibility or a function of the fingerprint API that doesn't work. Honestly, I already have trouble correcting the cases that fully doesn't work because I can't reproduce the bugs. But here it works half, I really don't know how I will solve it.

Linked to https://github.com/Kunzisoft/KeePassDX/issues/123
https://github.com/Kunzisoft/KeePassDX/issues/144
https://github.com/Kunzisoft/KeePassDX/issues/127
https://github.com/Kunzisoft/KeePassDX/issues/114

**Comment by jxosborne** [comment]
Thanks, I'll have a look through the bugs.  Its very easily reproducible, so maybe I'll see if I can capture some more debug data for it and get it figured out myself.

**Comment by gradinaruvasile** [comment]
@jxosborne 

Try grepping the logcat output for "keepass".
https://github.com/Kunzisoft/KeePassDX/issues/114

**Comment by aksh1618** [comment]
I'm also facing this on Android Pie on Redmi Note 4 SD (mido). Logs:
```
E/com.kunzisoft.keepass.fingerprint.FingerPrintHelper(28246): Unable to encrypt data
E/com.kunzisoft.keepass.fingerprint.FingerPrintHelper(28246): javax.crypto.IllegalBlockSizeException
E/com.kunzisoft.keepass.fingerprint.FingerPrintHelper(28246):  at android.security.keystore.AndroidKeyStoreCipherSpiBase.engineDoFinal(AndroidKeyStoreCipherSpiBase.java:519)
E/com.kunzisoft.keepass.fingerprint.FingerPrintHelper(28246):  at javax.crypto.Cipher.doFinal(Cipher.java:2055)
E/com.kunzisoft.keepass.fingerprint.FingerPrintHelper(28246):  at com.kunzisoft.keepass.fingerprint.FingerPrintHelper.encryptData(FingerPrintHelper.java:174)
E/com.kunzisoft.keepass.fingerprint.FingerPrintHelper(28246):  at com.kunzisoft.keepass.password.PasswordActivity$6.onAuthenticationSucceeded(PasswordActivity.java:662)
E/com.kunzisoft.keepass.fingerprint.FingerPrintHelper(28246):  at android.hardware.fingerprint.FingerprintManager$AuthenticationCallback.onAuthenticationSucceeded(FingerprintManager.java:257)
E/com.kunzisoft.keepass.fingerprint.FingerPrintHelper(28246):  at android.hardware.fingerprint.FingerprintManager.sendAuthenticatedSucceeded(FingerprintManager.java:954)
E/com.kunzisoft.keepass.fingerprint.FingerPrintHelper(28246):  at android.hardware.fingerprint.FingerprintManager.access$600(FingerprintManager.java:67)
E/com.kunzisoft.keepass.fingerprint.FingerPrintHelper(28246):  at android.hardware.fingerprint.FingerprintManager$MyHandler.handleMessage(FingerprintManager.java:893)
E/com.kunzisoft.keepass.fingerprint.FingerPrintHelper(28246):  at android.os.Handler.dispatchMessage(Handler.java:106)
E/com.kunzisoft.keepass.fingerprint.FingerPrintHelper(28246):  at android.os.Looper.loop(Looper.java:193)
E/com.kunzisoft.keepass.fingerprint.FingerPrintHelper(28246):  at android.app.ActivityThread.main(ActivityThread.java:6739)
E/com.kunzisoft.keepass.fingerprint.FingerPrintHelper(28246):  at java.lang.reflect.Method.invoke(Native Method)
E/com.kunzisoft.keepass.fingerprint.FingerPrintHelper(28246):  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:495)
E/com.kunzisoft.keepass.fingerprint.FingerPrintHelper(28246):  at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
E/com.kunzisoft.keepass.fingerprint.FingerPrintHelper(28246): Caused by: android.security.KeyStoreException: Key user not authenticated
E/com.kunzisoft.keepass.fingerprint.FingerPrintHelper(28246):  at android.security.KeyStore.getKeyStoreException(KeyStore.java:839)
E/com.kunzisoft.keepass.fingerprint.FingerPrintHelper(28246):  at android.security.keystore.KeyStoreCryptoOperationChunkedStreamer.
E/com.kunzisoft.keepass.fingerprint.FingerPrintHelper(28246):  at android.security.keystore.KeyStoreCryptoOperationChunkedStreamer.doFinal(KeyStoreCryptoOperationChunkedStreamer.java:217)
E/com.kunzisoft.keepass.fingerprint.FingerPrintHelper(28246):  at android.security.keystore.AndroidKeyStoreCipherSpiBase.engineDoFinal(AndroidKeyStoreCipherSpiBase.java:506)
E/com.kunzisoft.keepass.fingerprint.FingerPrintHelper(28246):  ... 13 more
```

**Comment by J-Jamet** [comment]
Fixed in 2.5.0.0beta23
